plugins/linux-display/fu-linux-display-plugin.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-linux-display-plugin.h"
+
+struct _FuLinuxDisplayPlugin {
+	FuPlugin parent_instance;
+	GPtrArray *devices;
+};
+
+G_DEFINE_TYPE(FuLinuxDisplayPlugin, fu_linux_display_plugin, FU_TYPE_PLUGIN)
+
+static FuDisplayState
+fu_linux_display_plugin_get_display_state(FuLinuxDisplayPlugin *self)
+{
+	FuDisplayState display_state = FU_DISPLAY_STATE_DISCONNECTED;
+
+	/* no devices detected */
+	if (self->devices->len == 0)
+		return FU_DISPLAY_STATE_UNKNOWN;
+
+	/* any connected display is good enough */
+	for (guint i = 0; i < self->devices->len; i++) {
+		FuDevice *device = g_ptr_array_index(self->devices, i);
+		const gchar *status =
+		    fu_udev_device_get_sysfs_attr(FU_UDEV_DEVICE(device), "status", NULL);
+		if (g_strcmp0(status, "connected") == 0) {
+			display_state = FU_DISPLAY_STATE_CONNECTED;
+			break;
+		}
+	}
+	return display_state;
+}
+
+static void
+fu_linux_display_plugin_ensure_display_state(FuLinuxDisplayPlugin *self)
+{
+	FuContext *ctx = fu_plugin_get_context(FU_PLUGIN(self));
+	if (!fu_plugin_has_flag(FU_PLUGIN(self), FWUPD_PLUGIN_FLAG_READY))
+		return;
+	fu_context_set_display_state(ctx, fu_linux_display_plugin_get_display_state(self));
+}
+
+static gboolean
+fu_linux_display_plugin_plugin_backend_device_added(FuPlugin *plugin,
+						    FuDevice *device,
+						    FuProgress *progress,
+						    GError **error)
+{
+	FuLinuxDisplayPlugin *self = FU_LINUX_DISPLAY_PLUGIN(plugin);
+	if (!fu_udev_device_set_physical_id(FU_UDEV_DEVICE(device), "drm", error))
+		return FALSE;
+	g_debug("device added: %s", fu_device_get_id(device));
```

Comment:
There aren't any IDs specified at the time this runs AFAICT.
```
01:26:53.280 FuEngine             /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1 added FuUdevDevice:
  Flags:                none
  BackendId:            /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1
  AcquiesceDelay:       2500
  PossiblePlugin:       linux_display
  Subsystem:            drm
  DeviceFile:           /dev/dri/card1
  SysfsPath:            /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1

Loading…                 [****************                       ]01:26:53.280 FuEngine             /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1 added FuUdevDevice:
  Flags:                none
  Guid[quirks]:         129bda59-932a-5210-b863-6f46f0a87584 ← DRM\TYPE_DRM_MINOR
  BackendId:            /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1
  AcquiesceDelay:       2500
  PossiblePlugin:       linux_display
  Subsystem:            drm
  DeviceFile:           /dev/dri/card1
  SysfsPath:            /sys/devices/pci0000:00/0000:00:08.1/0000:63:00.0/drm/card1

01:26:53.280 FuPlugin             backend_device_added(linux_display)
01:26:53.280 FuPluginLinuxDisplay device added: (null)
```

Reply:
Fixed, thanks.